src/target/ext/register_info.rs

```diff
@@ -0,0 +1,49 @@
+//! Provide register information for the target. See the lldb-gdb-remote [documentation](https://github.com/llvm/llvm-project/blob/main/lldb/docs/lldb-gdb-remote.txt) for more information.
+//!
+//! Some targets don't have register context in the compiled version of the debugger. Help the debugger by dynamically supplying the register info from the target if the packet 'qXfer:features:read' isn't used to provide a target description. The debugger will request the register info in a sequential manner till an 'E45' packet is received.
+//!
+//! _Note:_ LLDB specific! This packet is meant as an alternative to the 'qXfer:features:read:target.xml' packet. LLDB will fall back to the .xml file if qRegisterInfo isn't supported on the target. See issue [#99](https://github.com/daniel5151/gdbstub/issues/99) for more info on LLDB compatibility.
+//!
+use crate::target::Target;
+
+/// Describes the register info for a single register of the target
+pub struct Register<'a> {
+    /// The primary register name
+    pub name: &'a str,
+    /// An alternate name for the register
+    pub alt_name: &'a str,
+    /// Size in bits of a register (base 10)
+    pub bitsize: u16,
+    /// The offset within the 'g' and 'G' packet of the register data for this register (base 10)
+    pub offset: u16,
+    /// The encoding type of the register
+    pub encoding: &'a str,
+    /// The preferred format for display of this register
+    pub format: &'a str,
+    /// The register set name this register belongs to
+    pub set: &'a str,
+    /// The GCC compiler registers number for this register
+    pub gcc: Option<u16>,
+    /// The DWARF register number for this register that is used for this register in the debug information
+    pub dwarf: Option<u16>,
+    /// Specify as a generic register
+    pub generic: Option<&'a str>,
+    /// Other concrete register values this register is contained in (expecting base 10 here but will be written with write_num() as hex)
```

Comment:
no need to include protocol nitty gritty details in the help text here (i.e: the end user doesn't care if it's being written in decimal vs. hex)